modules/database/src/main/java/com/intuit/wasabi/database/impl/DBITransaction.java

```diff
@@ -178,78 +197,82 @@ private Object perform(HandleBlock block) {
 
             throw ex;
         } finally {
-            if (!inTransaction) {
-                close();
+            // Close the handle (internally this call releases the JDBC connection to the pool)
+            if (handle!=null) {
+                handle.close();
             }
         }
     }
 
+    /**
+     * Creates a new handle to perform database operation.
+     * Handle represents a connection to the database system. It is a wrapper around a JDBC Connection object.
+     *
+     * @return handle to perform database operation.
+     *
+     */
     protected Handle getHandle() {
-        if (null == handle) {
-            handle = dbi.open();
-            // Force MySQL timezone to UTC
-            handle.execute("set time_zone = \"+0:00\"");
-        }
+        //This operation reuses JDBC Connection object from the pool and then creates a new handle object.
+        Handle handle = dbi.open();
+
+        // Force MySQL timezone to UTC
+        handle.execute("set time_zone = \"+0:00\"");
 
         return handle;
     }
 
-    protected void begin() throws WasabiServerException {
-        Handle h = getHandle();
-
+    /**
+     * Begin a new transaction for this instance of DBITransaction.
+     *
+     * Initialize the handle of this instance of DBITransaction
+     *
+     * @throws WasabiServerException
+     */
+    protected void begin(Handle handle) throws WasabiServerException {
         try {
-            Connection connection = h.getConnection();
+            Connection connection = handle.getConnection();
 
             connection.setAutoCommit(false);
         } catch (SQLException e) {
             throw new DatabaseException("Error obtaining database connection", e);
         }
 
-        h.begin();
-
-        inTransaction = true;
+        handle.begin();
+        inTransaction.set(true);
     }
 
-    protected void commit() {
-        if (handleIsMissing()) {
-            return;
-        }
-
+    /**
+     * Commits an active transaction of this instance of DBITransaction
+     *
+     */
+    protected void commit(Handle handle) {
         handle.commit();
-
-        inTransaction = false;
+        inTransaction.set(false);
     }
 
-    protected void rollback() {
-        if (handleIsMissing()) {
-            return;
-        }
-
+    /**
+     * Rollback an active transaction of this instance of DBITransaction
+     *
+     */
+    protected void rollback(Handle handle) {
         handle.rollback();
-
-        inTransaction = false;
+        inTransaction.set(false);
     }
 
-    public void close() {
-        if (handleIsMissing()) {
-            return;
-        }
-
+    /**
+     * Close the handle (JDBC Connection) of this instance of DBITransaction and set the handle to NULL
```

Comment:
We are not setting the handle to NULL anymore. Can we please remove it from the comment.